astrocut/cutout_processing.py

```diff
@@ -0,0 +1,429 @@
+    
+        cards = []
+        for att in ['TTYPE', 'TFORM', 'TUNIT', 'TDISP', 'TDIM']:
+            try:
+                cards.append(table_headers[0].cards[att+num])
+            except KeyError:
+                pass  # if we don't have info for this keyword, just skip it
+        
+        column_info[colname] = (num, cards)
+
+    # Adding column descriptions and additional info
+    for kwd in new_header:
+        if "TTYPE" not in kwd:
+            continue
+        
+        colname = new_header[kwd]
+        num = kwd.replace("TTYPE", "")
+    
+        info_row = column_info.get(colname)
+        if not info_row:
+            new_header.comments[kwd] = 'column name'
+            new_header.comments[kwd.replace("TTYPE", "TFORM")] = 'column format'
+            continue
+    
+        info_num = info_row[0]
+        cards = info_row[1]
+    
+        for key, val, desc in cards:
+            key_new = key.replace(info_num, num)
+            try:
+                ext_card = new_header.cards[key_new]
+            
+                if ext_card[1]:
+                    val = ext_card[1]
+                if ext_card[2]:
+                    desc = ext_card[2]
+                
+                new_header[key_new] = (val, desc)
+            except KeyError:  # card does not already exist, just add new one
+                new_header.set(key_new, val, desc, after=kwd)
+
+    # Adding any additional keywords from the original cutout headers
+    shared_keywords = _combine_headers(table_headers, constant_only=True)
+    for kwd in shared_keywords:
+        if kwd in new_header:  # Don't overwrite anything already there
+            continue
+
+        if any(x in kwd for x in ["WCA", "WCS", "CTY", "CRP", "CRV", "CUN",
+                                  "CDL", "11PC", "12PC", "21PC", "22PC"]):  # Skipping column WCS keywords
+            continue
+
+        new_header.append(shared_keywords.cards[kwd])
+
+
+
+def center_on_path(path, size, cutout_fles, target=None, img_wcs=None,
+                   target_pixel_file=None, output_path=".", verbose=True):
+    """
+    Given a moving target path that crosses through one or more cutout files 
+    (as produced by `cube_cut`/tesscut) and size, create a target pixel file 
+    containint a cutout of the requested size centered on the moving target 
+    given in the providedpath.
+    
+    Note: No resampling is done so there will be some jitter in the moving target
+    placement.
+    
+    Parameters
+    ----------
+    path : `~astropy.table.Table`
+        Table (or similar) object with columns "time," containing `~astropy.time.Time`
+        objects and "position," containing `~astropy.coordinate.Skycoord` objects.
+    size : array
+        Size in pixels of the cutout rectangle centered on the path locations ther will be 
+        returned. Formatted as [ny,nx]
+    cutout_fles : list
+        List of strings, Target Pixel File paths that the path crosses.
```

Comment:
Isn't the input list of files is the Full Frame Image files?

Reply:
Ah, never mind. I see now that this is already working on the cutout files

Reply:
Nope, this expects normal cutout target pixel files.